src/DataCollection.Shared/Models/FeaturePopupCache.cs

```diff
@@ -0,0 +1,75 @@
+﻿/*******************************************************************************
+  * Copyright 2021 Esri
+  *
+  *  Licensed under the Apache License, Version 2.0 (the "License");
+  *  you may not use this file except in compliance with the License.
+  *  You may obtain a copy of the License at
+  *
+  *  https://www.apache.org/licenses/LICENSE-2.0
+  *
+  *   Unless required by applicable law or agreed to in writing, software
+  *   distributed under the License is distributed on an "AS IS" BASIS,
+  *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  *   See the License for the specific language governing permissions and
+  *   limitations under the License.
+******************************************************************************/
+
+using Esri.ArcGISRuntime.Data;
+using Esri.ArcGISRuntime.Mapping.Popups;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace Esri.ArcGISRuntime.OpenSourceApps.DataCollection.Shared.Models
+{
+    /// <summary>
+    /// Represents a global cache of PopupManagers used to enable selection from existing values for
+    /// a related table (e.g. to show a popup list of available species for a tree record).
+    /// </summary>
+    public class FeaturePopupCache
+    {
+        /// <summary>
+        /// Static reference is available for shared use
+        /// </summary>
+        public static FeaturePopupCache Instance = new FeaturePopupCache();
+
+        /// <summary>
+        /// Stores cached query results by table.
+        /// </summary>
+        public Dictionary<FeatureTable, IEnumerable<PopupManager>> PopupManagersByTable = new Dictionary<FeatureTable, IEnumerable<PopupManager>>();
+
+        /// <summary>
+        /// Performs the query, populating the cache with an ordered list of popup managers, each manager representing a feature in the table.
+        /// </summary>
+        public async Task PopulateCache(FeatureTable table)
+        {
+            if (PopupManagersByTable.ContainsKey(table))
```

Comment:
In a DM you mentioned, 

> I don’t think there’s a good way to only query changed /new features, but maintaining the cache with a manual refresh option would be reasonable I think

It doesn't appear there is a manual refresh option in place here. Should we be expecting one?